### PR TITLE
Reader sidebar - allow default selections for expanding submenu stream groups.

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
Related to # https://github.com/Automattic/wp-calypso/issues/97993

## Proposed Changes

Allows submenu groups in the reader sidebar to select a default selection when clicked, and separates toggle expanded/collapsed functionality to the up/down arrow icon.

* Adds `expandableIconClick` prop to the expandable heading component. When present, this wraps the expandable icon in a button element with that used as its onClick method.
* Updates the various menu groups in the reader to pass `onClick` and `expandableIconClick` props.
    * Selecting the "Red" section (seen below in testing instructions), should expand the menus but never collapse them, and should navigate to the default selection for that menu group.
    * Selecting the "Purple" section should always toggle expand/collapse with no change in stream selection.

Note - these are slightly outdated gifs - an old iteration of functionality but close enough to give you an idea of the changes.
BEFORE
reader-selection-before

AFTER
reader-selection-after

## Why are these changes being made?

* Reduce clicks required for reader users navigating between streams.

## Testing Instructions

Go to the reader:
* Test the various expandable menu groups in the reader sidebar: Recent, Lists, Tags, Orgs (A8C, P2, etc.).
* Verify that:
    * Clicking Region 1 (Red) of a collapsed menu both expands the menu and selects the first item.
        * For 0 lists this should load the create list page, for 0 tags this should load the /tags page. 
    * Clicking Region 1 (Red) of an expanded menu will only reselect the first item in the group, and not toggle the menu collapsed state.
    * Clicking Region 2 (Purple) will toggle the expanded menu opened/closed without any selection change in any case.
    * Verify these all work as expected through keyboard flows. Region 2 should now have an accessible focus outline style - this should be seen when navigating to use it with keyboard navigation, but should not be seen when clicking.
* Put your design hat or user empathy hat on, consider whether if this is an improvement and if the interactions make sense. Or if we should rethink the rules of conditional selection. 

Go to the site sidebar:
* Go to my home for a site in calypso.
* Verify the sidebar menu items for menu groups (Appearance, users, etc.) continue to work as they do today and have no changes.
* Smoke test between default/classic calypso views.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by Addison-Stavlo** [comment]
On the surface it feels good. But the edge cases im not as sure about.

For example, I wonder about this kind of case: say we are on an unrelated stream and go to click “Recent” here where it is expanded. 



Currently on the PR it just collapses that recent menu without selecting anything.

Alternatively, I wonder if it should not collapse the menu in that case and apply the default selection instead. Maybe “Recent” should only collapse the menu if one of its children is selected? or not at all? 🤔

**Comment by davemart-in** [comment]
Yeah, thanks for highlighting this.

Let's simplify it. Let's 100% remove the collapse functionality from the left (red side):

image

The only thing that will collapse it is if it's expanded and you click the purple dropdown portion.

I think it will, but does that resolves your edge case?

**Comment by Addison-Stavlo** [comment]
> I think it will, but does that resolves your edge case?

I guess thats a matter of opinion. The edge cases will be there regardless, I think we need to determine the best approach and most acceptable trade offs in general for it.

> Let's 100% remove the collapse functionality from the left (red side): The only thing that will collapse it is if it's expanded and you click the purple dropdown portion.

The downside here is we then have a button that stops doing anything under certain conditions. If im understanding that case, click "Recent" would:
* If its not expanded, it will expand the area.
* If not selected, it will select the default.
* If its expanded and selected, the button becomes inert. (no collapse functionality)

Is that button being inert at times a UX concern? (Im honestly not sure, maybe its fine). Do we see that as better than the current state where clicking "Recent" would:
* Toggle collapsed/expanded either way.
* Select the default when expanding, if the group is not yet selected.

**Comment by davemart-in** [comment]
Yeah...

You're right.

I'm tempted to say let's just close this out.

**Comment by robertbpugh** [comment]
It seems like having the key functionality be reachable in one click would be more important than the edge cases you mentioned. I will see if I can test it.

**Comment by robertbpugh** [comment]
Thanks for the testing help. My thoughts:

- Having Recent list posts in one click seems like a win to me. I imagine this is the most or second-most clicked item in the Reader sidebar and we're cutting clicks in half to accomplish what you likely wanted to do.
- I agree it's bad to have the expander Recent section, click on Recent, and have it collapse. I like Dave's proposed solution to have a click on the word Recent do nothing, and the arrow collapses the section. Feels intuitive to me, and I'd prefer that edge case vs. continuing with 2 clicks to view Recent.

**Comment by davemart-in** [comment]
This latest change feels nice @Addison-Stavlo.

One thing I noticed was the selected state is different between Recent and the other dropdowns in the left nav. Recent uses bold and the others don't. Thought on making it bold consistently for all of them?

**Comment by Addison-Stavlo** [comment]
> This latest change feels nice @Addison-Stavlo.

Im confused how you tested them because I think the calypso build is still building. 😅

**Comment by Addison-Stavlo** [comment]
> One thing I noticed was the selected state is different between Recent and the other dropdowns in the left nav. Recent uses bold and the others don't. Thought on making it bold consistently for all of them?

I haven't touched those styles at all here. Im partial to the blue myself, but maybe better to handle separately. Would we change the individual items selected to bold instead of blue as well?

Screenshot 2025-02-11 at 2 57 46 PM

**Comment by Addison-Stavlo** [comment]
The live branch should be up now. Im wondering if clicking "Recent" when it is already expanded and has a child selected should either:
A) Do nothing (intert button), current behavior and why we would still need to remove cursor style in this condition.
OR
B) Select the default selection for that group again.
🤔

**Comment by davemart-in** [comment]
> Im confused how you tested them because I think the calypso build is still building. 😅

Yeah, i didn't have the latest... :-) 

> I haven't touched those styles at all here. Im partial to the blue myself

Sorry, I think we're talking about two different things. I'm talking about the sub menu active links:

CleanShot 2025-02-11 at 15 49 01@2x

vs. 

CleanShot 2025-02-11 at 15 49 25@2x

It feels weird that they're styled differently. I like the bold one better. The blue link doesn't look selected to me.

> Clicking "Recent" (or that area on any of the collapsable) should:
> If its not expanded, it will expand the area.
> If not selected, it will select the default.
> If its expanded and selected, the button becomes inert. (no collapse functionality)
> Clicking the arrow on the right should always toggle without selection.

This seems to work. To Rob's point, it remedies the double click issue. 

> Im wondering if clicking "Recent" when the menu is already expanded and has one of its items currently selected should either:
> A) Do nothing (intert button), current behavior and why we would still need to remove cursor style in this condition.
OR
> B) Re-select the default selection for that group. (So in case of "Recent", clicking "Recent" when one of the sites in Recent is selected would change the selection back to "All")

Yeah, I like B.

**Comment by Addison-Stavlo** [comment]
> Sorry, I think we're talking about two different things. I'm talking about the sub menu active links: It feels weird that they're styled differently. I like the bold one better. The blue link doesn't look selected to me.

Lets please take this to a separate issue as it is a separate concern and want to keep the discussion and comment trail in the context of what we are changing here. Im talking about the same thing, blue is the selected state for regular items and the selected state of regular items inside submenus - imo keeping blue is consistent. Moving to bold makes us have selected items look 2 diferent ways depending where they are nested which seems off.

**Comment by robertbpugh** [comment]
Tests great to me! I also like B.

> B) Re-select the default selection for that group. (So in case of "Recent", clicking "Recent" when one of the sites in Recent is selected would change the selection back to "All")

**Comment by davemart-in** [comment]
> Lets please take this to a separate issue

Sounds smart. Works for me.

**Comment by Addison-Stavlo** [comment]
Thank you both for testing and giving your input! I will clean this up and apply "option B" later today. 😄

**Comment by Addison-Stavlo** [comment]
Not yet ready for testing... finding some weird issues

**Review by davemart-in — APPROVED** [review]
This feels solid to me. Thank you for being so methodical about this one. I like where we ended up.

File: client/reader/sidebar/reader-sidebar-tags/index.jsx [hunk 1]
```diff
@@ -43,6 +44,15 @@ export class ReaderSidebarTags extends Component {
 		this.setState( ( state ) => ( { addTagCounter: state.addTagCounter + 1 } ) );
 	};
 
+	selectMenu = () => {
+		const { onClick, tags, isOpen } = this.props;
+		if ( ! isOpen ) {
+			onClick();
+		}
+		const defaultSelection = tags?.length ? `/tag/${ tags[ 0 ]?.slug }` : '/tags';
+		page.redirect( defaultSelection );
```
Comment (Addison-Stavlo) [inline client/reader/sidebar/reader-sidebar-tags/index.jsx]: I will update this to use `page` instead of `page.redirect` and only call it if we aren't on that route. This should help with browser history working more as expected. But shouldn't change any other functionality.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17251346

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Addison-Stavlo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17251346).
